code/modules/hydroponics/seed_controller.dm

```diff
@@ -0,0 +1,123 @@
+// Attempts to offload processing for the spreading plants from the MC.
+// Processes vines/spreading plants.
+
+#define PLANTS_PER_TICK 500 // Cap on number of plant segments processed.
+#define PLANT_TICK_TIME 50  // Number of ticks between the plant processor cycling.
+
+// Debug for testing seed genes.
+/client/proc/show_plant_genes()
+	set category = "Debug"
+	set name = "Show Plant Genes"
+	set desc = "Prints the round's plant gene masks."
+
+	if(!holder)	return
+
+	if(!plant_controller || !plant_controller.gene_tag_masks)
+		usr << "Gene masks not set."
+		return
+
+	for(var/mask in plant_controller.gene_tag_masks)
+		usr << "[mask]: [plant_controller.gene_tag_masks[mask]]"
+
+var/global/datum/controller/plants/plant_controller // Set in New().
+
+/datum/controller/plants
+
+	var/plants_per_tick = PLANTS_PER_TICK
+	var/plant_tick_time = PLANT_TICK_TIME
+	var/list/plant_queue = list()           // All queued plants.
+	var/list/seeds = list()                 // All seed data stored here.
+	var/list/gene_tag_masks = list()        // Gene obfuscation for delicious trial and error goodness.
```

Comment:
@Probe1 actually, since the plant controller isn't an actual process under the process scheduler, this won't happen.

Though eventually (maybe not in this PR) it should be changed to one.